File: cs50x/week6/pset6/sentimental-readability/readability.py
# TODO
def main():
    text = input("Text: ")
    # Calculate words, letters and sentences
    num_letters = 0
    num_words = text.count(" ") + 1
    num_sentences = 0
    for i in text:
        if i == "." or i == "?" or i == "!":
            num_sentences += 1
        if i.isalpha():
            num_letters += 1
    # Calculate variables for CLI
    L = num_letters / num_words * 100
    S = num_sentences / num_words * 100
    cli = round(0.0588 * L - 0.296 * S - 15.8)
    # Output based on cli
    if cli >= 16:
        print("Grade 16+")
    elif cli < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {cli}")


if __name__ == "__main__":
    main()

File: cs50x/week2/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>
#include <ctype.h>

void getGrade(string input);
int calcLetters(string input);
int calcWords(string input);
int calcSentences(string input);

int main(void)
{
    string input = get_string("Text: ");
    getGrade(input);
}
// Coleman-Liau index
void getGrade(string input)
{
    // calc letters, words and sentences
    int letters = calcLetters(input);
    int words = calcWords(input);
    int sentences = calcSentences(input);
    // average num of letters per 100 words
    double L = ((float)letters / (float)words) * 100;
    // average num of sentences per 100 words
    double S = ((float)sentences / (float)words) * 100;
    // Coleman-Liau Algorithm
    double index = index = 0.0588 * L - 0.296 * S - 15.8;
    // index rounded
    index = round(index);
    // output depending on index
    if (index >= 16)
    {
        index = 16;
        printf("Grade %i+\n", (int)index);
    }
    else if (index < 1)
    {
        index = 1;
        printf("Before Grade %i\n", (int)index);
    }
    else
    {
        printf("Grade %i\n", (int)index);
    }
}
// calc letters
int calcLetters(string input)
{
    int result = 0;
    for (int i = 0, length = strlen(input); i < length; i++)
    {
        if (isalnum(input[i]))
        {
            result++;
        }
    }
    return result;
}
// calc words
int calcWords(string input)
{
    int result = 0;
    for (int i = 0, length = strlen(input); i < length; i++)
    {
        if (isspace(input[i]))
        {
            result++;
        }
    }
    result++;
    return result;
}
// calc sentences
int calcSentences(string input)
{
    int result = 0;
    for (int i = 0, length = strlen(input); i < length; i++)
    {
        if (input[i] == '!' || input[i] == '.' || input[i] == '?')
        {
            result++;
        }
    }
    return result;
}